lxd-agent/metrics.go

```diff
@@ -372,6 +379,13 @@ func getMemoryMetrics() (metrics.MemoryMetrics, error) {
 		}
 	}
 
+	// Calculate RSS using the simpler and more modern approach
+	if foundMemTotal && foundMemAvailable {
+		// Formula: RSS = MemTotal - MemAvailable
+		// This is how modern tools like 'free' calculate used memory
+		out.RSSBytes = out.MemTotalBytes - out.MemAvailableBytes
+	}
```

Comment:
I think we can leverage the default value of 0 and use it to detect if `RSSByte` needs to be set to non-0:
```suggestion
	if out.MemTotalBytes > out.MemAvailableBytes {
		// Modern tools like 'free' define RSS as being equal to: MemTotal - MemAvailable
		out.RSSBytes = out.MemTotalBytes - out.MemAvailableBytes
	}
```

This way, we wouldn't need the  2 booleans.

Reply:
@simondeziel  Thank you for the excellent suggestion! I've implemented your approach to simplify the code. Sorry for the late update, work picked up on my end.